src/Slider/_tests_/index.test.js

```diff
@@ -0,0 +1,488 @@
+// @flow
+
+import * as React from "react";
+import { mount, shallow } from "enzyme";
+
+import { PureSlider as Slider } from "../index";
+
+describe("Slider", () => {
+  beforeEach(() => {
+    // $FlowExpected
+    Element.prototype.getBoundingClientRect = jest.fn(() => {
+      return {
+        width: 400,
+        height: 0,
+        top: 0,
+        left: 28,
+        bottom: 0,
+        right: 428,
+      };
+    });
+  });
+  const onChange = jest.fn();
+  const onChangeBefore = jest.fn();
+  const onChangeAfter = jest.fn();
+  const component = mount(
+    <Slider onChange={onChange} onChangeBefore={onChangeBefore} onChangeAfter={onChangeAfter} />,
+  );
+  const instance = component.instance();
+  test("calculateValue should return proper value", () => {
+    expect(instance.calculateValue(0.52, false, false)).toEqual(52);
+    expect(instance.calculateValue(0.52, true, false)).toEqual(53);
+    expect(instance.calculateValue(0.52, false, true)).toEqual(51);
+    expect(instance.calculateValue(0.52, true, true)).toEqual(52);
+  });
+  test("calculateValueFromPosition should return proper value", () => {
+    expect(instance.calculateValueFromPosition(52)).toEqual(7);
+    expect(instance.calculateValueFromPosition(352)).toEqual(81);
+    expect(instance.calculateValueFromPosition(400)).toEqual(93);
+    expect(instance.calculateValueFromPosition(428)).toEqual(100);
+  });
+  test("findClosestKey should return proper key", () => {
+    expect(instance.findClosestKey(1, [1, 12])).toEqual(0);
+    expect(instance.findClosestKey(2, [1, 12])).toEqual(0);
+    expect(instance.findClosestKey(6, [1, 12])).toEqual(0);
+    expect(instance.findClosestKey(7, [1, 12])).toEqual(1);
+    expect(instance.findClosestKey(11, [1, 12])).toEqual(1);
+    expect(instance.findClosestKey(11, [1, 12, 24])).toEqual(1);
+    expect(instance.findClosestKey(20, [1, 12, 24])).toEqual(2);
+  });
+  test("moveValueByStep should return proper value", () => {
+    expect(instance.moveValueByStep(1)).toEqual(2);
```

Comment:
if for some reason this test runs after `"alignValueToStep should return proper value"`, then it will fail because the `step` size would not allow to get to `2`.

Maybe it would be a good idea to reset the `instance` before each test.